tests/src/python/providertestbase.py

```diff
@@ -451,7 +451,7 @@ def testExtentSubsetString(self):
             provider_extent = self.source.extent()
             self.source.setSubsetString(None)
             self.assertEqual(count, 0)
-            self.assertTrue(provider_extent.isNull())
+            self.assertTrue(provider_extent.isEmpty())
```

Comment:
But I don't think this should be the case

Reply:
I think it should be null for an empty layer too, can you change that please?

Reply:
Rebased to master and removed this commit, I'll have to find out why the tests end up getting an empty but not null extent

Reply:
Yeah, I'd say the provider is wrong there

Reply:
I get our friend 0,0,0,0 empty rect instead of the null one:

> AssertionError: <QgsRectangle: 0 0, 0 0> != <QgsRectangle: 0 1, 2 3>

Still trying to figure out why

Reply:
Ok figured and fixed, confirming by merging [PR 54917 ](https://github.com/qgis/QGIS/pull/54917) into this branch. If you can please approve that one I'll merge and rebase this after dealing with the same issue with the AFS provider.